src/js/port_handler.js

```
@@ -289,14 +305,20 @@ PortHandler.updateDeviceList = async function (deviceType) {
+                }
                 break;
             case "usb":
-                ports = await WEBUSBDFU.getDevices();
+                if (this.showUsbOption) {
+                    ports = await WEBUSBDFU.getDevices();
+                }
                 break;
             case "serial":
+                if (this.showSerialOption) {
+                    ports = await serial.getDevices("serial");
+                }
```

Biome (1.9.4)

[error] This case is falling through to the next case.

Add a `break` or `return` statement to the end of this case to prevent fallthrough.

(lint/suspicious/noFallthroughSwitchClause)